nautobot/dcim/tests/test_views.py

```diff
@@ -2223,6 +2223,19 @@ def setUpTestData(cls):
             "controller_managed_device_group": ControllerManagedDeviceGroup.objects.first().pk,
         }
 
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+    def test_vdc_panel_includes_add_vdc_btn(self):
+        """Assert Add Virtual device Contexts button is in Device detail view: Issue from #6348"""
+        device = Device.objects.first()
+        url = reverse("dcim:device", kwargs={"pk": device.pk})
+        response = self.client.get(url)
+        response_context = response.content.decode(response.charset)
+        vdc_url = reverse("dcim:virtualdevicecontext_add")
+        return_url = device.get_absolute_url()
+        vdcs_table_add_url = f"{vdc_url}?device={device.id}&amp;return_url={return_url}"
+        self.assertIn("Add virtual device contexts", response_context)
+        self.assertIn(vdcs_table_add_url, response_context)
```

Comment:
What about `assertInHtml` and test if there is an `<a href=` element with proper url?

Reply:
Tried using `assertInHtml` I got this error: 
```python
AssertionError: Second argument is not valid HTML:
('Unexpected end tag `td` (Line 254, Column 40)', (254, 40))
```

Reply:
I didn't saw the update code nor output, but probably there is some invalid HTML code either in assert or in the output HTML.